src/io/InputNode.hpp

```diff
@@ -0,0 +1,36 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Lab
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Lab
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_INPUTNODE_HPP
+#define QMCPLUSPLUS_INPUTNODE_HPP
+
+#include <memory>
+
+namespace qmcplusplus
+{
+
+/** Interface to a class to be type erased for the purposes of ownership
+ *  To allow a vector of unique_ptr to InputNode to contain a heterolist of dynamic objects
+ */
+struct InputNode
+{
+  virtual ~InputNode(){};
+  template<typename T, typename VOWN, typename VINPUT, typename... Args>
+  static void append(VOWN& vown, VINPUT& vinput, Args&&... args)
```

Comment:
It doesn't seem to be related to `InputNode` at all.
Should vown be of type `std::vector<std::unique_ptr<InputNode>>` explicitly?